eduNEXT/eox-tenant
# perf: filter certificates by organization and remove certificate proxy DS-305

## Description

This PR corresponds to the feature [JU-3](https://docs.google.com/document/d/1BtEShTyy5I9g_noIX0vA70dp_Oq4dCqKjCUdQN9xM5o/edit?usp=sharing). This adds a filter to render certificates by organizations.

Changes:
- Adds the pipeline step filter and its tests.
- Remove all about to Certificates Proxy.
    - The proxy setup in tenant_wise/__init__ and a method that is only used for that proxy.
    - The CertificateManager, ElegibleCertificateManager, and the GeneratedCertificateProxy.
    - The Certificate Proxy setting in common.
 - Remove all about the certificates backend (settings, tests, and the backend).
 - Fix quality test.

## Testing instructions

### In stage

- Enter in stage (Ex: https://tenant-a.nutmeg-live.edunext.link/courses)
- Create an account.
- Enroll in Course Tenant A 2 or Course Test Certificate 2.
- If the milestone works well, you are going to need to pass the first version of that course.
- Enroll in the first version of that course and pass it.
- Go to progress and request your certificate.
- You are able to see your certificate, but if you change the tenant you can't see your certificate.
- Then you can start Course Tenant A 2 or Course Test Certificate 2.

The last two steps test whether the feature works correctly.

### In local

Requisites
- A tutor env with this eox-tenant version.
    - You can add this in the config.yml:
```
OPENEDX_EXTRA_PIP_REQUIREMENTS:
- git+https://github.com/eduNEXT/eox-tenant.git@nu/ednx/ju-3#egg=eox-tenant
```
- You need a tenant config with  `"EDNX_USE_SIGNAL": true` and  `"OPEN_EDX_FILTERS_CONFIG"` defined. And have a tenant route with that config. Example of tenant:
```
{
    "EDNX_USE_SIGNAL": true,
    "OPEN_EDX_FILTERS_CONFIG": {
        "org.openedx.learning.certificate.render.started.v1": {
            "fail_silently": false,
            "pipeline": [
                "eox_tenant.filters.pipeline.FilterRenderCertificatesByOrg"
            ]
        }
    },
    "PLATFORM_NAME": "Tenant A",
    "SITE_NAME": "tenant-a.nutmeg.edunext.link",
    "course_org_filter": [
        "tenantA"
    ]
}
```

- You need a course with certificates.
    - Create a course with the tenant course_org_filter in Studio.
    - In LMS Admin, set Honor that course.
    - In Studio > Course > Settings > Schedule, change the start date.
    - In Studio > Course > Settings > Grading, set Homework total grade 100, total number 1, and dropable 0. Delete the rest.
    - In Studio > Content > Outline create a section, subsection, and unit with a problem. Then in the settings of the subsection, and add the previous step grading.
    - In Admin LMS > Certificates > Certificate generation configurations, create a enable element.
    - In Studio > Settings > Advanced click over Show Deprecated Settings and set Certificates Downloadable Before End to true.
    - In Studio > Settings > Certificates, create a certificate and activate it. (If you want to confirm, click Live Preview)
    - With an admin account in LMS > Course > Instructor > Certificate, enable the student to generate the certificate.

- Note: I don't know if it is necessary but I also added a enable element in Admin LMS > Grades > Persistent grades enabled flag.

Step by Step to verify the filter

- Create a user in the tenant.
- Enroll the user in the course.
- Pass the course.
- In Progress generate the certificate.
- View the certificate.
- Try to view that certificate with another tenant. (other-tenant/certificates/id-certificate) - You can't see it.

Step by Step to verify the milestones

- Create another course.
- Make the first one a requirement for the new one (In Studio > Settings > Scheduling > Prerequisite)
- The milestone will be generated on the admin site.
- Complete the prerequisite course, and a user milestone must be generated.

More info: [JU-3](https://docs.google.com/document/d/1BtEShTyy5I9g_noIX0vA70dp_Oq4dCqKjCUdQN9xM5o/edit?usp=sharing)

## Checklist for Merge
